src/components/donation/Steps/DonateMethods/PayPal.tsx

```diff
@@ -0,0 +1,36 @@
+import {
+  PayPalButtons,
+  PayPalScriptProvider,
+  ReactPayPalScriptOptions,
+  usePayPalScriptReducer,
+} from "@paypal/react-paypal-js";
+import { useErrorContext } from "contexts/ErrorContext";
+import LoaderRing from "components/LoaderRing";
+import { PAYPAL_CLIENT_ID } from "constants/env";
+
+const initialOptions: ReactPayPalScriptOptions = {
+  clientId: PAYPAL_CLIENT_ID,
+  commit: true,
+  currency: "USD",
+  enableFunding: "card,paylater,venmo",
+};
+
+export default function PayPal() {
+  return (
+    <PayPalScriptProvider options={initialOptions}>
+      <ButtonWrapper />
+    </PayPalScriptProvider>
+  );
+}
+
+function ButtonWrapper() {
+  const [{ isPending }] = usePayPalScriptReducer();
+  const { handleError } = useErrorContext();
+
+  return (
+    <div className="grid place-items-center min-h-[16rem] isolate">
+      {isPending && <LoaderRing thickness={10} classes="w-8" />}
+      <PayPalButtons className="w-full max-w-xs" onError={handleError} />
```

Comment:
Note: switching to mobile view and opening the PayPal popup and then closing it before it loads causes an error:
![image](https://github.com/AngelProtocolFinance/angelprotocol-web-app/assets/19427053/68ac844e-eb57-4cb1-8446-f403d1d64f1e)

This was allegedly addressed in this PR https://github.com/paypal/paypal-checkout-components/issues/132, but the problem reoccurred and it being tracked in the repo of the popup implementation PayPal uses called `zoid` https://github.com/krakenjs/zoid/issues/430. 

The most elegant way to handle this is to treat this as an error and handle it the same way we handle other errors.
Although far from optimal, the number of times a user would close the PayPal popup before it loads is small enough.